src/main/java/com/google/sps/servlets/GoDataServlet.java

```diff
@@ -14,108 +14,84 @@
-    ScavengerHunt hunt = buildScavengerHunt();
+    String huntID = request.getParameter(Constants.HUNTID_PARAMETER);
+    Entity huntEntity = findScavengerHunt(huntID);
+    if (huntEntity == null) {
+      return;
+    }
 
     response.setContentType(Constants.JSON_TYPE);
-    Gson gson = new Gson();
-    String json = gson.toJson(hunt);
+    String json = (String) huntEntity.getProperty(Constants.HUNT_VAL);
     response.getWriter().println(json);
   }
 
-  private ScavengerHunt buildScavengerHunt() {
-    // Constructing the first HuntItem.
-    Riddle firstRiddle =
-        new Riddle.Builder()
-            .withPuzzle("I was constructed in 1933")
-            .withHint("I am at the periphery of SF")
-            .withHint("I am golden in color")
-            .build();
-    LatLng firstCoord = new LatLng.Builder().withLat(37.819).withLng(-122.479).build();
-    HuntItem firstHunt =
-        new HuntItem.Builder()
-            .withName("Golden Gate Bridge")
-            .atLocation(firstCoord)
-            .withDescription("A famous bridge in San Francisco")
-            .withRiddle(firstRiddle)
-            .build();
-
-    // Constructing the second HuntItem.
-    Riddle secondRiddle =
-        new Riddle.Builder()
-            .withPuzzle("A famous tower in Paris")
-            .withHint("I am very tall")
-            .withHint("I am a popular tourist destination")
-            .build();
-    LatLng secondCoord = new LatLng.Builder().withLat(48.858).withLng(2.295).build();
-    HuntItem secondHunt =
-        new HuntItem.Builder()
-            .withName("Eiffel Tower")
-            .atLocation(secondCoord)
-            .withDescription("A famous tower in Paris")
-            .withRiddle(secondRiddle)
-            .build();
-
-    // Constructing the third HuntItem.
-    Riddle thirdRiddle =
-        new Riddle.Builder()
-            .withPuzzle("I am often associated with America")
-            .withHint("I am a statue")
-            .withHint("I am very tall")
-            .build();
-    LatLng thirdCoord = new LatLng.Builder().withLat(40.689).withLng(-74.045).build();
-    HuntItem thirdHunt =
-        new HuntItem.Builder()
-            .withName("Statue of Liberty")
-            .atLocation(thirdCoord)
-            .withDescription("A statue in New York City")
-            .withRiddle(thirdRiddle)
-            .build();
-
-    // Constructing the scavenger hunt.
-    ArrayList<HuntItem> items = new ArrayList<HuntItem>();
-    items.add(firstHunt);
-    items.add(secondHunt);
-    items.add(thirdHunt);
-    ScavengerHunt hunt = new ScavengerHunt(items, index, "San Francisco");
-    return hunt;
+  /**
+   * Retrieves the scavenger hunt from Datastore using {@code huntID}, the ID corresponding to the
+   * scavenger hunt that should be retrieved.
+   */
+  private Entity findScavengerHunt(String huntID) {
+    Query huntQuery = new Query(Constants.SCAVENGER_HUNT_ENTITY);
```

Comment:
Instead of retrieving all entities and then filtering for the key, can you set up the Query to already be looking for this key?

https://cloud.google.com/datastore/docs/concepts/queries#key_filters